tests/TestCase/Database/ConnectionTest.php

```diff
@@ -962,4 +962,27 @@ public function testSchemaCollection()
         $connection->schemaCollection($schema);
         $this->assertSame($schema, $connection->schemaCollection());
     }
+
+    /**
+     * Tests supportsCrossWith
+     *
+     * @return void
+     */
+    public function testSupportsCrossWith()
+    {
+        $connection = new Connection(ConnectionManager::config('test'));
+        $targetConnection = new Connection(ConnectionManager::config('test'));
+
+        $this->assertFalse($connection->supportsCrossWith($targetConnection), 'The same connection can\'t used in cross');
```

Comment:
There's no real harm in that, its just that it would cause every 'normal' query to use the new cross schema expression: https://github.com/cakephp/cakephp/pull/8354/files#diff-29389a5b2b355bb26375b37d7b5c3c44R530